internal/loader/configuration_client_manager.go

```diff
@@ -461,3 +474,59 @@ func Jitter(duration time.Duration) time.Duration {
 	// Apply the random jitter to the original duration
 	return duration + time.Duration(randomJitter)
 }
+
+func newClientAssertionCredential(ctx context.Context, serviceAccountName string, serviceAccountNamespace string) (azcore.TokenCredential, error) {
+	cfg, err := ctrlcfg.GetConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	client, err := client.New(cfg, client.Options{})
+	if err != nil {
+		return nil, err
+	}
+
+	serviceAccountObj := &corev1.ServiceAccount{}
+	err = client.Get(ctx, types.NamespacedName{Namespace: serviceAccountNamespace, Name: serviceAccountName}, serviceAccountObj)
+	if err != nil {
+		return nil, err
+	}
+
+	clientId := serviceAccountObj.ObjectMeta.Annotations[AnnotationClientID]
+	tenantId := serviceAccountObj.Annotations[AnnotationTenantID]
+	getAssertionFunc := newGetAssertionFunc(serviceAccountNamespace, serviceAccountName)
+
+	clientAssertionCredential, err := azidentity.NewClientAssertionCredential(tenantId, clientId, getAssertionFunc, nil)
```

Comment:
I am not sure the usage of `clientAssertionCredential`,for authenticating both service principal and workload identity?